File: yl-common-old/src/main/java/com/yoloo/server/common/vo/Validatable.java
package com.yoloo.server.common.vo;

import java.util.Set;
import javax.validation.ConstraintViolation;
import javax.validation.ConstraintViolationException;
import javax.validation.Validation;

public interface Validatable {

  default void validate() {
    Set<ConstraintViolation<Validatable>> violations =
        Validation.buildDefaultValidatorFactory().getValidator().validate(this);

    if (!violations.isEmpty()) {
      throw new ConstraintViolationException(violations);
    }
  }
}

File: src/main/kotlin/com/yoloo/server/core/common/Validatable.kt
package com.yoloo.server.core.common

import com.googlecode.objectify.annotation.OnSave
import javax.validation.ConstraintViolationException
import javax.validation.Validation

interface Validatable {

    @OnSave
    fun validate() {
        val violations = Validation.buildDefaultValidatorFactory().validator.validate(this)

        if (violations.isNotEmpty()) {
            throw ConstraintViolationException(violations)
        }
    }
}